File: examples/python/core/parallel/basic-usage.py
from aws_durable_execution_sdk_python import (
    BatchResult,
    DurableContext,
    durable_execution,
)

@durable_execution
def handler(event: dict, context: DurableContext) -> list[str]:
    """Execute three tasks in parallel."""
    # Define functions to execute in parallel
    task1 = lambda ctx: ctx.step(lambda _: "Task 1 complete", name="task1")
    task2 = lambda ctx: ctx.step(lambda _: "Task 2 complete", name="task2")
    task3 = lambda ctx: ctx.step(lambda _: "Task 3 complete", name="task3")
    
    # Execute all tasks concurrently
    result: BatchResult[str] = context.parallel([task1, task2, task3])
    
    # Return successful results
    return result.get_results()

File: examples/java/core/parallel/basic-usage.java
public class BasicParallelHandler extends DurableHandler<Map<String, Object>, List<String>> {

    @Override
    public List<String> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Define async branches to execute in parallel
        var future1 = ctx.runInChildContextAsync("task1", String.class,
            child -> child.step("task1", String.class, stepCtx -> "Task 1 complete"));

        var future2 = ctx.runInChildContextAsync("task2", String.class,
            child -> child.step("task2", String.class, stepCtx -> "Task 2 complete"));

        var future3 = ctx.runInChildContextAsync("task3", String.class,
            child -> child.step("task3", String.class, stepCtx -> "Task 3 complete"));

        // Execute all tasks concurrently
        DurableFuture.allOf(future1, future2, future3);

        return List.of(future1.get(), future2.get(), future3.get());
    }
}
